src/Header/HttpListener.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_HEADER_HTTPLISTENER_H_
+#define GROUP4_HEADER_HTTPLISTENER_H_
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <memory.h>
+
+#include <thread>
+
+#include "MainManager.h"
+
+class MainManager
+{
+};
+
+//Listen specific port and will callback MainManager function when receive request
+class HttpListener
+{
+
+public:
+    //Singleton code convention, must bind callback for listening port and require listening port
+    static std::unique_ptr<HttpListener> GetInstance(void (*callbackFunction)(std::string), const int &portNum);
+    //bind specific port to listen
+    void BindPort(const int &portNum);
+    //using multithread to listen
+    void StartListen();
+
+    //non copyable
+    HttpListener(HttpListener &cpy) = delete;
+    HttpListener &operator=(const HttpListener &cpy) = delete;
+
+private:
+    //Singleton code convention
+    HttpListener();
+
+    //will be used when callback.
+    void (*listenCallback)(std::string);
```

Comment:
Consider using std::function.
Also, consider adding a setter for this instead of putting it in the initializer.
You can consider using std::optional for this variable, and assert(listen_callback.has_value())

Reply:
Done